webui/react/src/stores/projects.tsx

```diff
@@ -49,6 +53,96 @@ class ProjectStore {
       return projects ? Loaded(projects) : NotLoaded;
     });
   }
+
+  public deleteProject(id: number) {
```

Comment:
now that this can get updates out of band, consider using [`ImmutableObservable`](https://github.com/determined-ai/determined/blob/main/webui/react/src/utils/observable.ts#L25) along with `asValueObj`  to prevent rerenders